src/recorder/jamrecorder.cpp

```diff
@@ -102,6 +102,32 @@ void CJamClient::Disconnect()
     wavFile = nullptr;
 }
 
+/**
+ * @brief CJamClient::TranslateChars Replace non-ASCII chars with nearest equivalent, if any
+ */
+QString CJamClient::TranslateChars (const QString& input) const
+{
+    static const char charmap[128] = {
+        '_', '_', ',', 'f', '"', '_', '_', '_', '^', '%', 'S', '<', 'E', '_', 'Z', '_',
+        '_', '\'', '\'', '"', '"', '_', '-', '-', '~', '_', 's', '>', 'e', '_', 'z', 'Y',
+        ' ', '_', '_', '_', '_', 'Y', '|', '_', '_', '_', 'a', '"', '-', '_', '_', '_',
+        '_', '_', '_', '_', '_', 'u', '_', '.', ',', '_', 'o', '"', '_', '_', '_', '_',
+        'A', 'A', 'A', 'A', 'A', 'A', 'E', 'C', 'E', 'E', 'E', 'E', 'I', 'I', 'I', 'I',
+        'D', 'N', 'O', 'O', 'O', 'O', 'O', 'x', 'O', 'U', 'U', 'U', 'U', 'Y', 'P', 'S',
+        'a', 'a', 'a', 'a', 'a', 'a', 'e', 'c', 'e', 'e', 'e', 'e', 'i', 'i', 'i', 'i',
+        'd', 'n', 'o', 'o', 'o', 'o', 'o', '_', 'o', 'u', 'u', 'u', 'u', 'y', 'p', 'y'
+    };
+
+    QByteArray r = input.toLatin1();
+
+    for (auto &c : r)
+    {
+        c = (c >= 128) ? charmap[c - 128] : (c >= 32) ? c : '_';
```

Comment:
Getting this with gcc 10.2.0:

```
src/recorder/jamrecorder.cpp: In member function 'QString recorder::CJamClient::TranslateChars(const QString&) const':
src/recorder/jamrecorder.cpp:125:16: warning: comparison is always false due to limited range of data type [-Wtype-limits]
  125 |         c = (c >= 128) ? charmap[c - 128] : (c >= 32) ? c : '_';
      |              ~~^~~~~~
```

In fact, German 'Ä' is replaced by '_' but I guess it was supposed to be replaced by 'A' in your logic?

Reply:
Ok, this must be a portability thing. On your system `auto &c` must be getting typed as `signed char`, whereas on mine it gets typed as `unsigned char`. That would be why on your system it never uses the array.

I could try it with `unsigned char &c` instead, but that might end up as a type mismatch. Or maybe just cast `c` to unsigned within the loop body.

No time on Sat to look at it, will do Sun or Mon. I see the Mac build failed too, so will have to see why.

Reply:
I’ve pushed a change that hopefully might be portable. Let’s see.

Reply:
No more warnings for me and `Ä` is proplery replaced with `A`. Looks good.